packages/eslint-plugin/src/rules/no-unnecessary-type-conversion.ts

```diff
@@ -0,0 +1,369 @@
+              {
+                messageId: 'unnecessaryTypeConversionSuggestion',
+                data: { type: 'string' },
+                fix: getWrappingFixer({
+                  ...wrappingFixerParams,
+                  wrap: expr => `${expr} satisfies string`,
+                }),
+              },
+            ],
+          });
+        }
+      },
+      CallExpression(node: TSESTree.CallExpression): void {
+        if (
+          node.callee.type === AST_NODE_TYPES.Identifier &&
+          node.arguments.length === 1
+        ) {
+          const getType = () =>
+            getConstrainedTypeAtLocation(services, node.arguments[0]);
+
+          const isBuiltInCall = (name: string) => {
+            if ((node.callee as TSESTree.Identifier).name === name) {
+              const scope = context.sourceCode.getScope(node);
+              const variable = scope.set.get(name);
+              return !variable?.defs.length;
+            }
+            return false;
+          };
+
+          if (
+            // eslint-disable-next-line @typescript-eslint/internal/prefer-ast-types-enum
+            (isBuiltInCall('String') &&
+              doesUnderlyingTypeMatchFlag(
+                getType(),
+                ts.TypeFlags.StringLike,
+              )) ||
+            (isBuiltInCall('Number') &&
+              doesUnderlyingTypeMatchFlag(
+                getType(),
+                ts.TypeFlags.NumberLike,
+              )) ||
+            // eslint-disable-next-line @typescript-eslint/internal/prefer-ast-types-enum
+            (isBuiltInCall('Boolean') &&
+              doesUnderlyingTypeMatchFlag(
+                getType(),
+                ts.TypeFlags.BooleanLike,
+              )) ||
+            (isBuiltInCall('BigInt') &&
+              doesUnderlyingTypeMatchFlag(getType(), ts.TypeFlags.BigIntLike))
+          ) {
+            const wrappingFixerParams = {
+              node,
+              innerNode: [node.arguments[0]],
+              sourceCode: context.sourceCode,
+            };
+            const typeString = node.callee.name.toLowerCase();
+
+            context.report({
+              loc: node.callee.loc,
+              messageId: 'unnecessaryTypeConversion',
+              data: {
+                type: node.callee.name.toLowerCase(),
+                violation: `Passing a ${typeString} to ${node.callee.name}()`,
+              },
+              fix: getWrappingFixer(wrappingFixerParams),
+              suggest: [
+                {
+                  messageId: 'unnecessaryTypeConversionSuggestion',
+                  data: { type: typeString },
+                  fix: getWrappingFixer({
+                    ...wrappingFixerParams,
+                    wrap: expr => `${expr} satisfies ${typeString}`,
+                  }),
+                },
+              ],
+            });
+          }
+        }
+      },
+      'CallExpression > MemberExpression.callee > Identifier[name = "toString"].property'(
```

Comment:
Today I learned that you could use ESQuery with the `.callee` like this. Cool!

Question, though, does the `.property` at the end do anything? I can't think what it would do and no tests fail if I change it to 

```suggestion
      'CallExpression > MemberExpression.callee > Identifier[name = "toString"]'(
```

Reply:
To make sure the selector is correct could you also add the following (valid) test case?
```ts
export {};
declare const toString: string;
toString.toUpperCase();
```

Reply:
That test fails without the `.property`, which I guess is what you had in mind? Adding the test, leaving the selector for that reason.